evalml/tests/automl_tests/test_automl.py

```diff
@@ -4488,6 +4464,77 @@ def test_cv_validation_scores_time_series(
     assert cv_vals[0] == validation_vals[0]
 
 
+@pytest.mark.parametrize("algorithm,batches", [("iterative", 2), ("default", 3)])
+@pytest.mark.parametrize(
+    "parameter,expected",
+    [
+        ("mean", ["mean", "median", "most_frequent"]),
+        (Categorical(["mean"]), Categorical(["mean"])),
+    ],
+)
+@pytest.mark.parametrize("problem_type", ["binary", "time series binary"])
+def test_search_parameters_held_automl(
```

Comment:
Can we also add coverage for a list-valued parameter specified at the top-level `search_parameters`? 

I think something like `"DateTime Featurizer": {"features_to_extract": ["month", "day_of_week"]}` would be an interesting case to cover for both algorithms. 

In this test or wherever you think is most appropriate.